CentralHub.Api/Services/SampleRoomRepository.cs

```diff
@@ -1,14 +1,18 @@
 using System.Collections.Immutable;
 using CentralHub.Api.Dtos;
+using CentralHub.Api.Threading;
 
 namespace CentralHub.Api.Services;
 
 public sealed class SampleRoomRepository : IRoomRepository
 {
-    private static readonly Dictionary<int, RoomDto> Rooms = new Dictionary<int, RoomDto>();
-    private static int _nextId;
+    private sealed class LockedStuff
+    {
+        public Dictionary<int, RoomDto> Rooms { get; } = new Dictionary<int, RoomDto>();
+        public int NextId { get; set; }
+    }
 
-    private static readonly object LockObject = new object();
+    private static readonly CancellableMutex<LockedStuff> LockedStuffMutex = new CancellableMutex<LockedStuff>(new LockedStuff());
```

Comment:
seems a little strange that you're locking two resources at the same time? are we never accessing one without accessing the other?

Reply:
We did that before. I think it makes sense to lock both at the same time as when we need to add a new room we need to increment NextId and we need to add it to the dictionary which requires both to be locked anyway.